plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -251,12 +251,15 @@ public function get_current_orders_pending_sync_count( $use_cache = false ): int
 		}
 
 		if ( $this->custom_orders_table_is_authoritative() ) {
-			$missing_orders_count_sql = "
+			$missing_orders_count_sql = $wpdb->prepare(
+				"
 SELECT COUNT(1) FROM $wpdb->posts posts
 INNER JOIN $orders_table orders ON posts.id=orders.id
 WHERE posts.post_type = '" . self::PLACEHOLDER_ORDER_POST_TYPE . "'
  AND orders.status not in ( 'auto-draft' )
-";
+ AND orders.type IN ($order_post_type_placeholder)",
+				$order_post_types
```

Comment:
Optional: Since we are adding the prepare and placeholder we could make this more readable by moving `self::PLACEHOLDER_ORDER_POST_TYPE` to a placeholder in both changed SQL statements.